### PR TITLE
feat: add startswith and endswith funcs

### PR DESCRIPTION
**Pull request by FooBartn**
Purpose: Add `startswith` and `endswith` built-in functions to Terraform

Per the issue referenced below, these are functions that the Terraform team is willing to accept due to the broad use case and interest. I have not submitted to the `terraform` repo before, so my apologies if I missed any conventions or standards. I tried to keep it as close to the other code that I saw as possible. If there's anything that should be changed please let me know.

Referencing Issue: https://github.com/hashicorp/terraform/issues/28209

### CODE REVIEW

**Comment by crw** [comment]
Thanks for this submission! It is in the queue to be reviewed. Thanks again!

**Review by alisdair — CHANGES_REQUESTED** [review]
This is great, thanks for the contribution! This behaviour makes sense to me and I checked it out locally.

I requested a few additional test cases inline. We'll also need documentation for these new functions, which means:

- [Adding a new page to the `language/functions` directory](https://github.com/hashicorp/terraform/tree/main/website/docs/language/functions), with the same format as other string functions
- [Adding the page to the `language-nav-data.json` file](https://github.com/hashicorp/terraform/blob/main/website/data/language-nav-data.json)
- [Updating the `layouts/language.erb` template](https://github.com/hashicorp/terraform/blob/main/website/layouts/language.erb)

I'm adding our Technical Writer as a reviewer on this PR, who should be able to help review the docs once you've written them.

**Comment by laurapacilio** [comment]
Hi there! I'm excited to see this new functionality, and thank you to @alisdair for pointing out the need for docs! :) I think a great approach to maintain the format we use for these pages is to copy and paste the .mdx from an existing string function page and then modify it for the new function that you're adding. 

I would love to review your docs changes once you've drafted them. If you have questions at any point in the process, please just leave a comment in this PR and tag me, and I'd be happy to help.

Thank you again for your contributions!

**Comment by FooBartn** [comment]
@alisdair Unsure what broke with Vercel. Let me know if there's anything I need to do.

@laurapacilio Please feel free tear apart my first run at documentation 😄

**Review by alisdair — COMMENTED** [review]
Thanks! I'll leave the docs review for Laura, but the code is ✅ from me.

I think the Vercel failure is because we don't have entries in the language-nav-data.json file at the top level of the `functions` part of the tree. Other functions have an entry for their virtual sub-section (which you've added) and a separate `hidden` entry at the `functions` level, later in the file. For example: https://github.com/hashicorp/terraform/blob/f74e93a84ab47f6e060ff124211adb38c7c8112f/website/data/language-nav-data.json#L861-L862

**Comment by FooBartn** [comment]
> I think the Vercel failure is because we don't have entries in the language-nav-data.json file at the top level of the `functions` part of the tree. Other functions have an entry for their virtual sub-section (which you've added) and a separate `hidden` entry at the `functions` level, later in the file. For example:

That was it! Thank you!

File: website/docs/language/functions/startswith.mdx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+true
+
+> startswith("hello world", "world")
+false
+```
+
+## Related Functions
+
+- [`endswith`](/language/functions/endswith) takes a given string and a given suffix value,
+  and returns true if the first given string contains the given value at its end
```
Comment (laurapacilio) [inline website/docs/language/functions/startswith.mdx:30]:
```suggestion
- [`endswith`](/language/functions/endswith) takes two values: a string to check and a suffix string. The function returns true if the first string ends with that exact suffix value.
```

File: website/docs/language/functions/endswith.mdx [hunk 2]
```diff
@@ -1,14 +1,12 @@
 ---
 page_title: endswith - Functions - Configuration Language
 description: |-
-  The endswith function takes a given string and a given suffix value,
-  and returns true if the first given string contains the given value at its end.
+  The endswith function takes two values: a string to check and a suffix string. It returns true if the first string ends with that exact suffix value.
```
Comment (laurapacilio) [inline website/docs/language/functions/endswith.mdx:4]:
```suggestion
  The endswith function takes two values: a string to check and a suffix string. It returns true if the first string ends with that exact suffix.
```

**Review by laurapacilio — APPROVED** [review]
Approving because I think this is great! Just a few final minor wording nits to remove an extra word. Would you mind merging those in as well?

Otherwise, @alisdair I think this is ready - I'd like you to do the honors, since this contains code as well as docs :)

**Comment by FooBartn** [comment]
> Approving because I think this is great! Just a few final minor wording nits to remove an extra word. Would you mind merging those in as well?
> 
> Otherwise, @alisdair I think this is ready - I'd like you to do the honors, since this contains code as well as docs :)

I like it! I went with language that was similar to another set of docs, but this is clearer. Thank you!

**Comment by alisdair** [comment]
Thanks again! 🎉

**Pull request merged**